@here/harp-mapview-decoder/lib/TileLoader.ts

```diff
@@ -79,159 +80,79 @@ export class TileLoader implements ITileLoader {
-     * @returns A promise which resolves the [[TileLoaderState]].
-     */
-    loadAndDecode(): Promise<TileLoaderState> {
-        switch (this.state) {
-            case TileLoaderState.Loading:
-            case TileLoaderState.Loaded:
-            case TileLoaderState.Decoding:
-                // tile is already loading
-                return this.donePromise!;
-
-            case TileLoaderState.Ready:
-            case TileLoaderState.Failed:
-            case TileLoaderState.Initialized:
-            case TileLoaderState.Canceled:
-                // restart loading
-                this.startLoading();
-                return this.donePromise!;
-        }
+        protected tileDecoder: ITileDecoder
+    ) {
+        super(dataSource, tileKey);
     }
 
     /**
-     * Return the current state in form of a promise. Caller can then wait for the promise to be
-     * resolved.
-     *
-     * @returns A promise which resolves the current [[TileLoaderState]].
+     * @override
      */
-    waitSettled(): Promise<TileLoaderState> {
-        if (!this.donePromise) {
-            return Promise.resolve(this.state);
-        }
-        return this.donePromise;
+    get priority(): number {
+        return this.m_priority;
     }
 
     /**
-     * Cancel loading of the [[Tile]].
-     * Cancellation token is notified, an internal state is cleaned up.
+     * @override
      */
-    cancel() {
-        switch (this.state) {
-            case TileLoaderState.Loading:
-                this.loadAbortController.abort();
-                this.loadAbortController = new AbortController();
-                break;
-
-            case TileLoaderState.Decoding:
-                if (this.requestController) {
-                    this.requestController.abort();
-                    this.requestController = undefined;
-                }
-                break;
-        }
-
-        this.onDone(TileLoaderState.Canceled);
-    }
-
-    /**
-     * Return `true` if [[Tile]] is still loading, `false` otherwise.
-     */
-    get isFinished(): boolean {
-        return (
-            this.state === TileLoaderState.Ready ||
-            this.state === TileLoaderState.Canceled ||
-            this.state === TileLoaderState.Failed
-        );
-    }
-
-    /**
-     * Update the priority of this [[Tile]]'s priority. Is effective to sort the decoding requests
-     * in the request queue (used during heavy load).
-     */
-    updatePriority(priority: number): void {
-        this.priority = priority;
+    set priority(priority: number) {
+        this.m_priority = priority;
         if (this.requestController !== undefined) {
             this.requestController.priority = priority;
         }
     }
 
     /**
-     * Start loading. Only call if loading did not start yet.
+     * @override
      */
-    protected startLoading() {
-        const myLoadCancellationToken = this.loadAbortController.signal;
+    protected loadImpl(
+        abortSignal: AbortSignal,
```

Comment:
this value is protected in the base class, so it can be exposed here, do you need to pass it in?

Reply:
Good point. I forgot to make the abort controller and all members related to the done promise private. I'll do that. That way the base class is as closed for modification as it's possible in Typescript.